src/renderers/dom/client/syntheticEvents/SyntheticEvent.js

```diff
@@ -156,13 +173,45 @@ assign(SyntheticEvent.prototype, {
    * `PooledClass` looks for `destructor` on each instance it releases.
    */
   destructor: function() {
+    var setNull = setToNullOrWarning.bind(this);
     var Interface = this.constructor.Interface;
-    for (var propName in Interface) {
-      this[propName] = null;
+    for (var interfacePropName in Interface) {
+      setNull(interfacePropName);
+    }
+    otherNullableProps.forEach(setNull);
+    this._destructored = true;
+
+    function setToNullOrWarning(propName) {
```

Comment:
I don't think we want to introduce a function (and bind it) in such a hot path such as event destructor. I think it would be better to hoist this code outside `destructor` and make sure that when `__DEV__` is `false`, *all* code and overhead related to this warning is removed.

Currently, in production, this would compile to 

```js
destructor: function() {
  // ...
  function setToNullOrWarning(propName) { // <-- function allocation
    this[propName] = null
  }
}
```

As you can see it’s a performance regression from just placing this code inline, which we want to avoid.